pkg/kritis/admission/admission.go

```diff
@@ -101,6 +101,24 @@ func handleDeployment(ar *v1beta1.AdmissionReview, admitResponse *v1beta1.Admiss
 		return err
 	}
 	glog.Infof("handling deployment %s...", deployment.Name)
+
+	operation := ar.Request.Operation
+	if operation == v1beta1.Update {
+		oldDeployment := appsv1.Deployment{}
+		if err := json.Unmarshal(ar.Request.OldObject.Raw, &oldDeployment); err != nil {
+			return err
+		}
+
+		// For UPDATE events, if there is no new image added, we can skip the check.
+		// This is required, so that DELETE events work for Deployment.
+		//
+		// Before deleting a deployment, kubernetes always make replicas to 0 which causes an
+		// UPDATE event.
+		if !hasNewImage(DeploymentImages(deployment), DeploymentImages(oldDeployment)) {
+			return nil
```

Comment:
I'd like to add INFO logging here to know it's skipped because of the same image.